cpp/ql/src/semmle/code/cpp/Type.qll

```diff
@@ -310,7 +326,17 @@ class ErroneousType extends BuiltInType {
 }
 
 /**
- * The unknown type.
+ * The unknown type.  This type has no corresponding C/C++ syntax.
+ *
+ * Unknown types usually occur inside _uninstantiated_ template functions.
+ * In the example below, the type of the ternary `?:` expression cannot
+ * be determined:
+ * ```
+ * template <typename T>
+ * T min(T x, T y) {
+ *   return (x < y) ? x : y;
+ * }
```

Comment:
If the `?:` expression doesn't have type `T`, that seems like a quirk of our extractor rather than a fundamental limitation. A better example would be 
```cpp
template<typename T>
bool check(T x) {
  if (x.a == x.b)
    abort();
}
```

I'd expect the types of `x.a` and `x.b` to be unknown. I'm less sure about the type of the possibly overloaded `==`, so let's not mention that.

Reply:
I agree, changed it to your example.